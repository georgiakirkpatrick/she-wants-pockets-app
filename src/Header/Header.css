* {
    box-sizing: border-box;
    padding: 0;
    margin: 0;
}

.Header {
    position: fixed;
    top: 0;
    left: 0;
    height: 56px;
    width: 100%;
    margin-block-start: 0;
    margin-block-end: 0;
    padding-inline-start: 0;
    font-family: 'Poppins', sans-serif;
    overflow-x: hidden;
}

.Header a {
    color: black;
}

.Header button {
    height: 100%;
    border: 0;
    background: #fff;
    color: #000;
    padding: 0.25rem 0.5rem;
    margin: 0.25rem;
    cursor: pointer;
}

.Header__hamburger div {
    width: 25px;
    height: 3px;
    margin: 5px;
    background-color: black;
    border-radius: 1px;
}

.Header__item {
    width: 25%;
    display: flex;
    flex-direction: row;
    align-items: center;
}

.Header__logo {
    justify-content: center;
    text-transform: uppercase;
    letter-spacing: 5px;
    font-size: 22px;
}

.Header__menu {
    display: none;
}

.Header__search-user {
    justify-content: flex-end;
}

.Header__search-user a {
    margin: 0 0 0 1em;
}

.Header__top {
    height: 10vh;
    width: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
}

@media screen and (min-width: 768px) {
    .Header__hamburger {
        display: none;
    }

    .Header__menu {
        height: 5vh;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: space-around;
        width: 100%;
        font-size: 18px;
    }

    .Header__hamburger {
        display: none;
    }

    /* .Header__nav-links {
        display: flex;
        flex-direction: row;
        justify-content: space-around;
    } */

    .Header__top {
        height: 5vh;
    }
}