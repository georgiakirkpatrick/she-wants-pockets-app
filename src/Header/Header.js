import React from 'react';
import './Header.css';
import '@fortawesome/react-fontawesome';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import '../FontAwesomeIcons/FontAwesomeIcons';

function Header(props) {
    return (
        // <Router>
        //     <nav>
        //         <ul>
        //             <li>
        //                 <Link to="/">Home</Link>
        //             </li>

        //             <li>
        //                 <Link to="/about">About</Link>
        //             </li>

        //             <li>
        //                 <Link to="/users">Users</Link>
        //             </li>
        //         </ul>
        //     </nav>

        //     <Switch>
        //         <Route path="/about">
        //             <About />
        //         </Route>

        //         <Route path="/users">
        //             <Users />
        //         </Route>

        //         <Route path="/">
        //             <Home />
        //         </Route>
        //     </Switch>
        // </Router>
        
        <nav className='Header'>
            <div className='Header__top'>
                <div className='Header__item'>
                        <button className='Header__hamburger' type='button' onClick={props.click} id='hamburger'>
                            <div className='line-1'></div>
                            <div className='line-2'></div>
                            <div className='line-3'></div>
                        </button>
                </div>
                
                <div className='Header__logo Header__item'>
                    <h4>
                        SWP
                    </h4>
                </div>
                
                <div className='Header__search-user Header__item'>
                    <button href='www.shewantspockets.com'>
                        <FontAwesomeIcon icon="search" size='lg' />
                    </button>

                    <button href='www.shewantspockets.com'>
                        <FontAwesomeIcon icon="user-circle" size='lg' />
                    </button>
                </div>
            </div>

            <ul className='Header__menu' id='menu'>
                <li><a href='/category'>Categories</a></li>
                <li><a href='/brand'>Brands</a></li>
                <li><a href='/principles'>Principles</a></li>
                <li><a href='/about'>About</a></li>
            </ul>
        </nav>
    )
}

export default Header