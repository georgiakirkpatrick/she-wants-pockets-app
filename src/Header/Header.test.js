import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import Header from './Header'

describe ('Header', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Router><Header /></Router>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Router><Header /></Router>)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})