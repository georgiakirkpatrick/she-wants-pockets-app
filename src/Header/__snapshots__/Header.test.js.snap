// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the UI as expected 1`] = `
<nav
  className="Header"
>
  <div
    className="Header__top"
  >
    <div
      className="Header__item"
    >
      <button
        className="Header__hamburger"
        id="hamburger"
        type="button"
      >
        <div
          className="line-1"
        />
        <div
          className="line-2"
        />
        <div
          className="line-3"
        />
      </button>
    </div>
    <div
      className="Header__logo Header__item"
    >
      <h4>
        SWP
      </h4>
    </div>
    <div
      className="Header__search-user Header__item"
    >
      <button
        href="www.shewantspockets.com"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-search fa-w-16 fa-lg "
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <button
        href="www.shewantspockets.com"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user-circle fa-w-16 fa-lg "
          data-icon="user-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 496 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
  <ul
    className="Header__menu"
    id="menu"
  >
    <li>
      <a
        href="/category"
      >
        Categories
      </a>
    </li>
    <li>
      <a
        href="/brand"
      >
        Brands
      </a>
    </li>
    <li>
      <a
        href="/principles"
      >
        Principles
      </a>
    </li>
    <li>
      <a
        href="/about"
      >
        About
      </a>
    </li>
  </ul>
</nav>
`;
